examples/gno.land/r/demo/gnolotto_factory/gnolotto_factory.gno

```diff
@@ -0,0 +1,239 @@
+package gnolotto_factory
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/gnolotto"
+	"gno.land/p/demo/ufmt"
+)
+
+var lotteries *avl.Tree
+
+// Replace this address with your address
+var admin std.Address = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+
+// Initializes the lottery AVL tree
+func init() {
+	lotteries = avl.NewTree()
+}
+
+// Creates a new lottery, only callable by admin.
+func CreateLottery(drawTime int64, prizePool int64) (int, string) {
+	sentCoins := std.GetOrigSend()
+	amount := sentCoins.AmountOf("ugnot")
+	banker := std.GetBanker(std.BankerTypeRealmSend)
+	send := std.Coins{{"ugnot", int64(amount)}}
+
+	if prizePool != amount {
+		banker.SendCoins(std.GetOrigPkgAddr(), std.GetOrigCaller(), send)
+		return -1, "Prize pool must match the transaction value"
+	}
+
```

Comment:
For this case, there is no need to manually return coins to the caller. You can simply call a `panic`, which will revert the whole transaction, which will revert any changed state, and also return any sent coins to their owner. Panic should be used for erroring in general, please apply this pattern for all errors in the realm. 

BTW- your use of `std.GetOrgPkgAddr()` seems incorrect. In this specific case, you are trying to send coins from the entry point (here, the transaction caller), to the caller himself. I assume you wanted to use `std.CurrentRealm().Addr()`

Reply:
Thank you for this clarification, I thought that the funds were sent even in the event of an error.

Reply:
You are correct - in case of an error returned, the coins will stay in the realm. However, if you panic, the whole state will be reverted, including the sent coins :)